front/app/components/Mounter/index.tsx

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react';
+
+interface Props {
+  onMount: () => void;
+}
+
+export default ({ onMount }: Props) => {
+  useEffect(() => {
+    setTimeout(() => {
+      onMount();
+    }, 200);
```

Comment:
The timeout makes sure the `onMount` function is called on the next tick. Without it, `onMount` is still called before `onData` is called by the components inside of the `Outlet` (tested this)

Reply:
Hm hm, doesn't sound ideal. Do you have the same issue if you make this functionality part of the Outlet?

Reply:
Probably

Reply:
This is a very common problem in js frameworks though- Vue has `this.$nextTick()`, and Svelte has `tick().then()`. It is weird React doesn't have this, so that's why you have to resort to `setTimeout`